.travis.yml

```diff
@@ -11,19 +6,27 @@ env:
 matrix:
   include:
     - os: osx
+      osx_image: xcode9.0
       language: node_js
       node_js: "8"
     - os: linux
-      language: generic
+      services: docker
+      # even though we don't build anything in the "docker" mode, it's best to define the language as node_js
+      # so that we can manipulate node_modules if needed and keep things consistent
+      language: node_js
+      node_js: 8
 
 cache:
   directories:
   - node_modules
   - $HOME/.cache/electron
   - $HOME/.cache/electron-builder
-
+  - $HOME/.npm
+  - $HOME/.yarn-cache
```

Comment:
No need to cache it. Only `node_modules` should be cached. Yarn is not NPM and uses existing node modules correctly. Because caching `.yarn-cache` will lead to gigantic cache (download and archive/unarchive time).

Reply:
I understand your point.
However, for .npm (I have no experience with Yarn) we felt the need to do this as we had to manually remove some files inside node_modules (they had some whitespaces), and keeping .npm makes things faster.
But I agree that these can be removed for most users.